### PR TITLE
Merge 'mapbox-gl-test-suite' repo without preserving git history

### LINKED ISSUE #3795: Merge the "mapbox-gl-test-suite" repo into this repo
ref #3656 

# Todo

 - [ ] move over files
 - [ ] investigate using `git-lfs` for all images & binary test fixtures 
 - [ ] migrate `mapbox-gl-test-suite` issues & PRs
 - [ ] configure `mapbox-gl-native` to pull test-suite files from `mapbox-gl-js` package
 - [ ] mark the `mapbox-gl-test-suite` github repo as deprecated
 - [ ] mark the `mapbox-gl-test-suite` npm package as deprecated

### PR DESCRIPTION
**Pull request by lucaswoj**
This PR merges `mapbox-gl-test-suite` into this repo. We do not preserve git history to reduce repo size. We introduce `git-lfs` to reduce repo size. 

The merge is pretty naive, dropping `mapbox-gl-test-suite` into a `test/integration` folder unmodified. More extensive merging of fixture libraries and code is planned for a future PR.

fixes #3795

### CODE REVIEW

**Comment by jfirebaugh** [comment]
Getting some errors when trying to use this in mapbox-gl-native:

```
~/Development/mapbox-gl-native $ npm install
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Downloading test/fixtures/mbsv5-6-18-23.vector.pbf (71.19 KB)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Error downloading object: test/fixtures/mbsv5-6-18-23.vector.pbf (d1b2db99907dbfd9747dc88e8ef5b91974b21222f6dde73f5507bec7e957545b)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Errors logged to /private/var/folders/vp/d4cnkmkx3pz5tj36vw4g5hbr0000gn/T/npm-84441-263c2371/git-cache-723ed10f/e68e3a6d5035d77f21dc15256369866cac7bfeab/.git/lfs/objects/logs/20161219T160701.103130306.log
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Use `git lfs logs last` to view the log.
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed 2
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: fatal: test/fixtures/mbsv5-6-18-23.vector.pbf: smudge filter lfs failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Downloading test/fixtures/mbsv5-6-18-23.vector.pbf (71.19 KB)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Error downloading object: test/fixtures/mbsv5-6-18-23.vector.pbf (d1b2db99907dbfd9747dc88e8ef5b91974b21222f6dde73f5507bec7e957545b)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Errors logged to /private/var/folders/vp/d4cnkmkx3pz5tj36vw4g5hbr0000gn/T/npm-84441-263c2371/git-cache-99b6f612/e68e3a6d5035d77f21dc15256369866cac7bfeab/.git/lfs/objects/logs/20161219T160808.428575012.log
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Use `git lfs logs last` to view the log.
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed 2
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: fatal: test/fixtures/mbsv5-6-18-23.vector.pbf: smudge filter lfs failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Downloading test/fixtures/mbsv5-6-18-23.vector.pbf (71.19 KB)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Error downloading object: test/fixtures/mbsv5-6-18-23.vector.pbf (d1b2db99907dbfd9747dc88e8ef5b91974b21222f6dde73f5507bec7e957545b)
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Errors logged to /private/var/folders/vp/d4cnkmkx3pz5tj36vw4g5hbr0000gn/T/npm-84441-263c2371/git-cache-9e6dd10d/e68e3a6d5035d77f21dc15256369866cac7bfeab/.git/lfs/objects/logs/20161219T160908.323343201.log
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: Use `git lfs logs last` to view the log.
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed 2
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: error: external filter git-lfs smudge %f failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: fatal: test/fixtures/mbsv5-6-18-23.vector.pbf: smudge filter lfs failed
npm ERR! git checkout e68e3a6d5035d77f21dc15256369866cac7bfeab: 
npm ERR! Darwin 15.6.0
npm ERR! argv "/usr/local/Cellar/node@6/6.9.1/bin/node" "/usr/local/bin/npm" "install"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! code 128
```

**Comment by lucaswoj** [comment]
I have a couple theories about the errors you're seeing. They could be caused by:

 - not having git-lfs installed on your machine 
 - npm not playing nice with git lfs
 - me force pushing a few small fixes to this PR

**Comment by lucaswoj** [comment]
@jfirebaugh pointed out some bugs when using `lfs` with npm & yarn

 - https://github.com/git-lfs/git-lfs/issues/899
 - https://github.com/npm/npm/issues/11151
 - https://github.com/yarnpkg/yarn/issues/1337

**Comment by lucaswoj** [comment]
I think I've found workable solutions to the CI and `npm install` challenges. Can you try again @jfirebaugh?

**Comment by lucaswoj** [comment]
Word of caution: opening this PR's "Files changed" tab on github crashes Chrome 😬

**Comment by jfirebaugh** [comment]
Can you remove the `actual.png`s? They shouldn't be checked in.

**Comment by lucaswoj** [comment]
Good catch @jfirebaugh. It looks like I updated `.gitignore` after adding the files to git. Fixed in the latest commit.

**Comment by jfirebaugh** [comment]
In my tests, git lfs is slower than our current mapbox-gl-test-suite setup on the cases where we currently suffer: fresh clones.

With git 2.11.0 and git-lfs 1.5.3, the initial checkout of this branch took 21m46.766s. It's true that git lfs caches the downloaded files, but this is a cost that anyone cloning the repository for the first time will have to pay. That includes installation of a git npm dependency, which does `git clone --mirror`.

In mapbox-gl-native, after removing the mapbox-gl-test-suite dependency and updating the mapbox-gl dependency to point at 68feb9c79219deab29014417136a254b0b48e536, a fresh `npm install` took a similar amount of time -- 22m53.584s. A non-fresh `npm install` after bumping the SHA of the `mapbox-gl` dev dependency from 68feb9c79219deab29014417136a254b0b48e536 to 65a49c37ae76311ef0dcbd109e6761bc512d669d (a noop commit) took 17m56.164s. For comparison, equivalent operations with the current setup take about 14m.

TL;DR, with npm, any time a git dependency changes, it's a fresh `git clone`, and that's slower with git lfs than cloning the entire history of mapbox-gl-test-suite. It doesn't seem like this workload is a good fit for git lfs.

**Comment by lucaswoj** [comment]
I agree. I'll strip out lfs in a subsequent commit.

cc @mourner

**Comment by lucaswoj** [comment]
Ready to 🚢?

**Comment by jfirebaugh** [comment]
Hold on this one please. I need to make sure it integrates cleanly into mapbox-gl-native.

**Comment by lucaswoj** [comment]
No problem 👍. Any idea how adding `expected.png` files and fixtures to `.npmignore` would effect GL Native?

**Comment by jfirebaugh** [comment]
It looks like that'll be an issue actually -- after installing as a dev dependency in GL native, `node_modules/mapbox-gl` doesn't have a `test` directory at all.

**Comment by jfirebaugh** [comment]
Looking promising downstream in gl-native: #7531. @lucaswoj could you update the branch here with the latest from test-suite?

**Comment by lucaswoj** [comment]
Updated to latest test suite SHA @jfirebaugh (https://github.com/mapbox/mapbox-gl-test-suite/commit/9ab19437e6a5272f1dfd7aaf12a7b5bbec208895)

**Comment by lucaswoj** [comment]
Merging per chat with @jfirebaugh 🎉
